content/docs/context.md

```diff
@@ -91,8 +91,12 @@ A more complex example with dynamic values for the theme:
 
 ## Consuming Multiple Contexts
 
+To keep context re-rendering fast, React needs to make each context consumer a separate node in the tree. 
```

Comment:
Why mention this? It's an implementation detail.

Reply:
https://github.com/reactjs/reactjs.org/pull/587/files#r176797917

Reply:
cc: @gaearon I feel like this is a rationalization, because we would likely use this same API even if it didn't happen to be good for performance. It's the way it is for multiple reasons, not only that one.

Reply:
Feel free to suggest an alternative :-) It's the only way I've been able to get people to put behind their preconception that the extra nesting is "ugly". I feel that if we don't double down on this and somehow motivate it (I don't care much how exactly), we'll keep getting the same confused looks.

Reply:
I suppose. Though the people who think the extra nesting is ugly aren't wrong :D

To me, the main motivation for using special component types is that it decouples context from the class API. So when we introduce a new component API in the future, context will already just work. This is one of the rationales behind higher-order component pattern we've been discussing.

Maybe we should just ship `useContext`. Or provide a recipe for it, since it's only a few lines.

But if you feel this sentence is a good way to head off objections, I'm cool with leaving it in, even if I don't agree it's the main reason we designed it this way.